packages/dag4-network/src/state-channel-token-network.ts

```diff
@@ -0,0 +1,111 @@
+import { StateChannelNetworkInfo } from "./types/network-info";
+import {
+  PostTransactionV2,
+  PendingTransaction,
+  TransactionV2,
+  SnapshotV2,
+} from "./dto/v2";
+import { StateChannelTokenL0Api } from "./api/state-channel-token/l0-api";
+import { StateChannelTokenL1Api } from "./api/state-channel-token/l1-api";
+
+class MethodNotAvailableError extends Error {
+  constructor(ctor: { new (...args: any[]): any }, method: Function) {
+    super(`${ctor.name}.${method.name} is not available`);
+  }
+}
+
+class StateChannelTokenNetwork {
```

Comment:
```suggestion
class L0TokenNetwork {
```

Reply:
Is there any reason to make this class reimplement all the same methods as the DAG network class? It's identical so I would think it would make more sense to extend the DagNetwork class to add any overrides needed.

Reply:
There are considerable differences between the two, especially in types. As DagNetwork supports both v1 and v2 dtos and is not generic it's difficult to work with types returned by the different methods. L0TokenNetwork is v2-compatible-only. In this case, I cite my knowledge and experience instead of opinion.